tool/test/org/antlr/v4/test/tool/BaseTest.java

```diff
@@ -112,6 +113,7 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+@SuppressWarnings("resource")
```

Comment:
Looks like we forgot to close the pipes on some code paths in `BaseTest.execClass`. We should fix that rather than use `@SuppressWarnings`. Either way it would best be done as it's own separate pull request. I have a branch ready that I created for testing, but since you originally found the issue I'll let you submit a pull request if you want to. Just let me know. :smile: